torchft/manager.py

```diff
@@ -288,8 +288,33 @@ def shutdown(self, wait: bool = True) -> None:
             self._manager.shutdown()
         self._executor.shutdown(wait=wait)
 
+    def collect_all_allreduce(
+        self, tensors: List[torch.Tensor], should_quantize: bool = False
+    ) -> torch.futures.Future[List[torch.futures.Future[torch.Tensor]]]:
```

Comment:
If we're doing a bulk operation we should consider using `allreduce_coalesced` instead of N individual collective operations and matching the naming scheme